parser/parser.go

```diff
@@ -1096,7 +1103,7 @@ func parseAddressValue(addressText string) (
 	var endOfUri int
 	var startOfParams int
 	if addressText[0] != '<' {
-		if displayName != nil {
+		if _, ok := displayName.(base.String); ok {
```

Comment:
It was one line shorter, and I was being lazy.  I'll change it to a switch, I think they're clearer.